heap_sort/HeapSort.java

```diff
@@ -0,0 +1,44 @@
+/*
+*  Following is the implementation of HeapSort algorithm
+*  It will sort the input array with time complexity O(N*logN)
+*  N being the size of array
+*/
+
+public class HeapSort {
+
+	public static void heapify(int[] arr, int size, int parent) {  // Build max-heap of subtree of parent index
+		int left = 2 * parent;
+		int right = left + 1;
+		int largest = parent;  // Initially considering parent as largest
+		if (left < size && arr[left] > arr[largest])  // Left child is greater than parent
+			largest = left;
+		if (right < size && arr[right] > arr[largest]) // Right child is greater than parent
+			largest = right;
+		if (largest != parent) {  // If parent is largest then subtree is max-heap
+			int temp = arr[parent];
+			arr[parent] = arr[largest];  // Swap largest child with parent
+			arr[largest] = temp;
+			heapify(arr, size, largest);  // Convert upper subtree to max-heap
+		}
+	}
+
+	public static void heapSort(int[] arr) {
+		for (int i = arr.length / 2 - 1; i>=0; i--)
+			heapify(arr, arr.length, i);  // Create max-heap
+
+		for (int i = arr.length - 1; i >= 0; i--) {
```

Comment:
If you use post-increment, it can require the compiler to have to generate code that creates an extra temporary variable. It wouldn't change your output btw.